code/backend/pages/02_Explore_Data.py

```diff
@@ -38,13 +38,21 @@
 try:
     search_handler = Search.get_search_handler(env_helper)
 
-    results = search_handler.search_client.search("*", facets=["title"])
-    unique_files = [filename["value"] for filename in results.get_facets()["title"]]
+    results = (
+        search_handler.search_client.search("*", facets=["title"])
+        if search_handler.search_client is not None
+        else None
+    )
+    unique_files = (
+        [filename["value"] for filename in results.get_facets()["title"]]
+        if results
+        else []
+    )
     filename = st.selectbox("Select your file:", unique_files)
     st.write("Showing chunks for:", filename)
 
     results = search_handler.perform_search(filename)
-    data = search_handler.process_results(results)
+    data = search_handler.process_results(results) if results else []
```

Comment:
Can this be handled inside search_handler.process_results?

Reply:
Updated